ui/pages/notifications/notification-components/snap/snap.tsx

```diff
@@ -1,32 +1,53 @@
-import React, { useContext } from 'react';
-import { useSelector, useDispatch } from 'react-redux';
+import React, { useContext, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import { useHistory } from 'react-router-dom';
+import { NotificationServicesController } from '@metamask/notification-services-controller';
 import {
   MetaMetricsEventCategory,
   MetaMetricsEventName,
 } from '../../../../../shared/constants/metametrics';
 import { MetaMetricsContext } from '../../../../contexts/metametrics';
 import { NotificationListItemSnap } from '../../../../components/multichain';
-import type { SnapNotification } from '../../snap/types/types';
 import { getSnapsMetadata } from '../../../../selectors';
-import { markNotificationsAsRead } from '../../../../store/actions';
 import { getSnapRoute, getSnapName } from '../../../../helpers/utils/util';
+import { useMarkNotificationAsRead } from '../../../../hooks/metamask-notifications/useNotifications';
+import { deleteExpiredSnapNotifications } from '../../../../store/actions';
 
 type SnapComponentProps = {
-  snapNotification: SnapNotification;
+  snapNotification: NotificationServicesController.Types.INotification;
 };
 
 export const SnapComponent = ({ snapNotification }: SnapComponentProps) => {
-  const dispatch = useDispatch();
   const history = useHistory();
+  const dispatch = useDispatch();
   const trackEvent = useContext(MetaMetricsContext);
+  const { markNotificationAsRead } = useMarkNotificationAsRead();
 
   const snapsMetadata = useSelector(getSnapsMetadata);
 
   const snapsNameGetter = getSnapName(snapsMetadata);
 
+  let timerId: NodeJS.Timeout | undefined;
+
+  const handleUnreadNotification = () => {
+    markNotificationAsRead([
+      {
+        id: snapNotification.id,
+        type: snapNotification.type,
+        isRead: snapNotification.isRead,
+      },
+    ]);
+
+    timerId = setTimeout(async () => {
```

Comment:
Before this change, the notification lifecycle was not behaving as one would expect. We were re-marking a notification as read on subsequent clicks and that was updating the `readAt` time which essentially extended the lifecycle of the notification. We also would not delete a notification until unmounting the notifications list, which again is not congruent with a lifecycle of 10 seconds. This timeout was added to ensure that the notification gets deleted when the user has remained in the notification list view AFTER click on a notification as read.